jax/_src/numpy/lax_numpy.py

```diff
@@ -1368,6 +1368,82 @@ def gradient(f, *varargs, axis: Optional[Union[int, Tuple[int, ...]]] = None,
 def isrealobj(x):
   return not iscomplexobj(x)
 
+@partial(jit, static_argnums=1)
+class _RankWarning(UserWarning):
+    """
+    Issued by `polyfit` when the Vandermonde matrix is rank deficient.
+    For more information, a way to suppress the warning, and an example of
+    `RankWarning` being issued, see `polyfit`.
+    """
+    pass
+
+@_wraps(np.polyfit)
+def polyfit(x, y, deg, rcond=None, full=False, w=None, cov=False):
+  """
+  Note: Unlike numpy, the function will not warn on rank reduction, which indicates an ill conditioned matrix
+  """
+  _check_arraylike("polyfit", x, y)
+  deg = core.concrete_or_error(int, deg, "deg must be int")
+  order = deg + 1
+  # check arguments.
+  if deg < 0:
+      raise ValueError("expected deg >= 0")
+  if x.ndim != 1:
+      raise TypeError("expected 1D vector for x")
+  if x.size == 0:
+      raise TypeError("expected non-empty vector for x")
+  if y.ndim < 1 or y.ndim > 2:
+      raise TypeError("expected 1D or 2D array for y")
+  if x.shape[0] != y.shape[0]:
+      raise TypeError("expected x and y to have same length")
+
+  # set rcond
+  if rcond is None:
+      rcond = len(x)*finfo(x.dtype).eps
+
+  # set up least squares equation for powers of x
+  lhs = vander(x, order)
+  rhs = y
+
+  # apply weighting
+  if w is not None:
+      w = _promote_dtypes_inexact(_check_arraylike("polyfit", w))
```

Comment:
I think `_check_arraylike` returns `None`, so this will have to be done in two lines.